server/src/dsl-utils.ts

```diff
@@ -2,90 +2,76 @@ import { errors } from "@openfga/syntax-transformer";
-          // Working, base line = 1, col = 1
-          if (e.metadata?.errorType) {
-            code = e.metadata.errorType;
-          }
-          return {
-            message: e.msg,
-            severity: DiagnosticSeverity.Error,
-            range: {
-              start: { line: linesStart - 1, character: charactersStart - 1 },
-              end: { line: lineEnd - 1, character: charactersEnd - 1 },
-            },
-            source: "ModelValidationError",
-            code,
-            data: { ...e.metadata, file: e.file },
-          };
-        } else if (e instanceof errors.ModuleTransformationSingleError) {
-          // Working on??
-          return {
-            message: e.msg,
-            severity: DiagnosticSeverity.Error,
-            range: {
-              start: { line: linesStart - 1, character: charactersStart - 1 },
-              end: { line: lineEnd - 1, character: charactersEnd - 1 },
-            },
-            source: "ModuleTransformationError",
-            code,
-            data: { ...e.metadata, file: e.file },
-          };
-        } else if (e instanceof errors.FGAModFileValidationSingleError) {
-          // Working, base line = 1, col = 1
-          const props = e.properties;
-          return {
-            message: props.msg,
-            severity: DiagnosticSeverity.Error,
-            range: {
-              start: { line: linesStart - 1, character: charactersStart - 1 },
-              end: { line: lineEnd - 1, character: charactersEnd - 1 },
-            },
-            source: "FGAModFileValidationError",
-            code,
-            metadata: { file: e.file },
-          };
-        } else {
-          throw new Error("Unhandled Exception: " + JSON.stringify(e, null, 4));
+      // Base line = 0, col = 0
+      if (e instanceof errors.DSLSyntaxSingleError) {
+        // Working, base line = 1, col = 0
+        return {
+          message: e.msg,
+          severity: DiagnosticSeverity.Error,
+          range: {
+            start: { line: linesStart - 1, character: charactersStart },
+            end: { line: lineEnd - 1, character: charactersEnd },
+          },
+          source: "SyntaxError",
+          code,
+          data: { ...e.metadata, file: e.file },
+        };
+      } else if (e instanceof errors.ModelValidationSingleError) {
+        // Working, base line = 1, col = 1
+        if (e.metadata?.errorType) {
+          code = e.metadata.errorType;
         }
-      },
-    ) || []
+        return {
+          message: e.msg,
+          severity: DiagnosticSeverity.Error,
+          range: {
+            start: { line: linesStart - 1, character: charactersStart - 1 },
+            end: { line: lineEnd - 1, character: charactersEnd - 1 },
+          },
+          source: "ModelValidationError",
+          code,
+          data: { ...e.metadata, file: e.file },
+        };
+      } else if (e instanceof errors.ModuleTransformationSingleError) {
+        // Working on??
```

Comment:
?

Reply:
Can you elaborate on this?

Reply:
Should be `// Working, base line = 1, col = 1`

Forgot to go back and investigate the line numbers:
https://github.com/openfga/language/blob/18b0254b2be1766f709cf4148cf46dcc2cb53407/pkg/js/util/exceptions.ts#L712-L719